rali/rali/source/randombboxcrop_reader.cpp

```diff
@@ -0,0 +1,323 @@
+        return 0;
+    }
+    auto it = _map_content.find(image_name);
+    if (_map_content.end() == it) 
+        THROW("ERROR: Given name not present in the map" + image_name)    
+    return it->second;
+}
+
+void RandomBBoxCropReader::add(std::string image_name, BoundingBoxCord crop_box)
+{
+    pCropCord random_bbox_cords = std::make_shared<CropCord>(crop_box.x, crop_box.y, crop_box.w, crop_box.h);
+    if (exists(image_name))
+    {
+        // auto it = _map_content.find(image_name);
+        return;
+    }
+    _map_content.insert(std::pair<std::string, std::shared_ptr<CropCord>>(image_name,random_bbox_cords));
+}
+
+void RandomBBoxCropReader::print_map_contents()
+{
+    pCropCord random_bbox_cords;
+
+    std::cerr << "\n ********************************Map contents:***************************** \n";
+    for (auto& elem : _map_content) {
+        std::cerr << "\n Name :\t " << elem.first;
+        random_bbox_cords = elem.second;
+        std::cerr<<"\n Crop values:: crop_x:: "<<random_bbox_cords->crop_x<<"\t crop_y:: "<<random_bbox_cords->crop_y<<"\t crop_width:: "<<random_bbox_cords->crop_width<<"\t crop_height:: "<<random_bbox_cords->crop_height;
+    }
+}
+
+void RandomBBoxCropReader::read_all()
+{
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    std::uniform_int_distribution<> dis(0, 6);
+
+    const std::vector<float> sample_options = {0.0f, 0.1f, 0.3f, 0.5f, 0.7f, 0.9f};
+    int sample_option;
+    std::pair<bool, float> option;
+    float min_iou; // max_iou;
+    bool invalid_bboxes = true;
+    _entire_iou = true;
+    bool _overlap_iou = false;
+    BoundingBoxCord crop_box, jth_box;
+    uint bb_count;
+    _meta_bbox_map_content = _meta_data_reader->get_map_content();
+    int i = 0;
+    for (auto& elem : _meta_bbox_map_content)
+    {
+        std::string image_name =  elem.first;
+        BoundingBoxCords bb_coords = elem.second->get_bb_cords();
+        ImgSizes img_sizes = elem.second->get_img_sizes();
+        in_width[i] = img_sizes[i].w;
+        in_height[i] = img_sizes[i].h;
+        bb_count = bb_coords.size();
+        std::vector<float> coords_buf(bb_count * 4);
+        for(unsigned int j = 0, m = 0; j < bb_count; j++, m += 4){
+            coords_buf[m] = bb_coords[j].x;
+            coords_buf[m + 1] = bb_coords[j].y;
+            coords_buf[m + 2] = bb_coords[j].w;
+            coords_buf[m + 3] = bb_coords[j].h;
+        } 
+        _crop_param->set_image_dimensions(in_width, in_height);
+        _crop_param->fill_crop_dims();
+        Parameter<float> *x_drift_factor = _crop_param->get_x_drift_factor();
+        Parameter<float> *y_drift_factor = _crop_param->get_y_drift_factor();
+        _x1_val = _crop_param->get_x1_arr_val();
+        _y1_val = _crop_param->get_y1_arr_val();
+        _crop_width_val = _crop_param->get_cropw_arr_val();
+        _crop_height_val = _crop_param->get_croph_arr_val();
+        bool crop_success = false;
+        int count = 0;
```

Comment:
move it to inside condition statement

Reply:
moved the count variable inside condition